tests/optimize/test_backtest_detail.py

```diff
@@ -593,14 +611,19 @@ def test_backtest_results(default_conf, fee, mocker, caplog, data) -> None:
 
     mocker.patch("freqtrade.exchange.Exchange.get_fee", return_value=0.0)
     mocker.patch("freqtrade.exchange.Exchange.get_min_pair_stake_amount", return_value=0.00001)
+    mocker.patch("freqtrade.exchange.Binance.get_max_leverage", return_value=100)
     patch_exchange(mocker)
     frame = _build_backtest_dataframe(data.data)
     backtesting = Backtesting(default_conf)
     backtesting._set_strategy(backtesting.strategylist[0])
     backtesting.required_startup = 0
+    if data.leverage > 1.0:
```

Comment:
Is data something that the strategy returns? The strategy can decide to open a short at 1x leverage.

Reply:
this is a ... different kind of test :laughing: 
you need to treat the whole file as "one" test (with parameters though)

data is simply a `BTContainer` object (an object helping define the backtesting testcase).
in real life, backtesting can define leverage, that's correct.
 
In this test/file - there is however no (real) strategy - only mocked data.
It's sole purpose is to test different behaviors in different situations during backtesting - most of which originating from one or the other issue where freqtrade behaved badly (speak - backtesting was not doing what it should, for example with trailing stop). 

This test helps in testing that behavior - as it'll allow for (pretty easy) test case creation (including the candles used), where to buy/sell and what profit to expect.